src/jxl.imageio/jxloutput.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: BSD-3-Clause and Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+#include <cassert>
+#include <cstdio>
+#include <vector>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <jxl/encode.h>
+#include <jxl/resizable_parallel_runner.h>
+
+OIIO_PLUGIN_NAMESPACE_BEGIN
+
+#define DBG if (0)
+
+
+// References:
+//  * https://jpegxl.info
+
+
+
+class JxlOutput final : public ImageOutput {
+public:
+    JxlOutput() { fprintf(stderr, "JxlOutput()\n"); init(); }
+    ~JxlOutput() override { close(); }
+    const char* format_name(void) const override { return "jxl"; }
+    int supports(string_view feature) const override
+    {
+        return (feature == "exif" || feature == "ioproxy");
+    }
+    bool open(const std::string& name, const ImageSpec& spec,
+              OpenMode mode = Create) override;
+    bool write_scanline(int y, int z, TypeDesc format, const void* data,
+                        stride_t xstride) override;
+    bool write_tile(int x, int y, int z, TypeDesc format, const void* data,
+                    stride_t xstride, stride_t ystride,
+                    stride_t zstride) override;
+    bool close() override;
+    bool copy_image(ImageInput* in) override;
+
+private:
+    std::string m_filename;
+
+    void init(void)
+    {
+        ioproxy_clear();
+        clear_outbuffer();
+    }
+
+    void clear_outbuffer()
+    {
+    }
+
+    void set_subsampling(const int components[])
+    {
+    }
+};
+
+
+
+OIIO_PLUGIN_EXPORTS_BEGIN
+
+OIIO_EXPORT ImageOutput*
+jxl_output_imageio_create()
+{
+    return new JxlOutput;
+}
+
+OIIO_EXPORT const char* jxl_output_extensions[]
+    = { "jxl", nullptr };
+
+OIIO_PLUGIN_EXPORTS_END
+
+
+
+bool
+JxlOutput::open(const std::string& name, const ImageSpec& newspec,
+                OpenMode mode)
+{
+    // Save name and spec for later use
+    m_filename = name;
+
+    return true;
```

Comment:
If these are not functional, they should all return false, since they are not succeeding at writing the file.

I think you may also want to consider just not having a JxlOutput or a jxloutput.cpp file at all unless/until you plan to implement JXL writing. It's totally reasonable to get the reader working and merged first, then come back to the writer. (Also, there are some formats that we want to be able to read, but care a lot less about being able to write, so not all readers have corresponding writers. Whether JXL falls into this category is a choice I'll leave to you.)